src/clusterfuzz/_internal/bot/fuzzers/syzkaller/runner.py

```diff
@@ -145,17 +198,77 @@ def repro(self, repro_timeout, repro_args):
     return engine.ReproduceResult(result.command, result.return_code,
                                   result.time_executed, result.output)
 
-  def fuzz(self,
-           fuzz_timeout,
-           additional_args,
-           unused_additional_args=None,
-           unused_extra_env=None):
+  def save_rawcover_output(self, pid: int):
+    """Find syzkaller port and write rawcover data to a file."""
+
+    port = find_syzkaller_port(pid)
+
+    # port not found
+    if not port:
+      logs.log_warn('Syzkaller port was not found')
+      return
+
+    rawcover = requests.get(f'http://localhost:{port}/rawcover').text
+
+    if not rawcover:
+      logs.log_warn('Syzkaller rawcover not yet loaded')
+      return
+
+    file_path = get_cover_file_path()
+    with open(file_path, 'w+') as f:
+      f.write(rawcover)
+      logs.log(f'Writing syzkaller rawcover to {file_path}')
+
+  def run_and_wait(self, syzkaller_args=None,
+                   timeout=None) -> new_process.ProcessResult:
+    process = self.run(syzkaller_args)
+
+    start_time = time.time()
+
+    pid = process.popen.pid
+
+    logs.log(f'Syzkaller pid = {pid}')
+
+    looping_timer = LoopingTimer(
+        RAWCOVER_RETRIEVE_INTERVAL,
+        self.save_rawcover_output,
+        args=[pid],
+    )
+    looping_timer.run()
+
+    if not timeout:
+      output = process.communicate()[0]
+      looping_timer.cancel()
```

Comment:
can you wrap lines 238-252 in a try/finally, and have `looping_timer.cancel()` as part of the `finally` block? 

Otherwise if we exception somewhere here and it's handled somewhere down the call stack, the timer thread will leak.

Reply:
didn't feel right about this `cancel` placement. I like this suggestion.